### ISSUE TITLE
Query latest models distinct by update_time

### ISSUE THREAD

**Issue by ribbon-xx** [realm/realm-java#4774]
#### Goal

> I'm converting a sqlite database to Realm and have a model that has fields are groupID and updated_time. I want to select all latest model in each group.

#### Expected Results

> I expect to get the list that contains all latest model sorted by updated_time DESC.

#### Actual Results

>  RealmResult.distinct() is deprecated, RealmQuery.distinct() cannot make sure result is latest.

#### Steps & Code to Reproduce

> I don't have data right now @@

#### Code Sample

```java

RealmResults<MyModel> realmResults = getRealm().where(MyModel.class)
					.notEqualTo(MyModel.GROUP_ID, 0)
					.notEqualTo(MyModel.IS_DELETED, 0)
					.findAllSorted(MyModel.LAST_UPDATE, Sort.DESCENDING);
			realmResults.distinct(MyModel.GROUP_ID); <<< Deprecated

```

#### Version of Realm and tooling
Realm version(s): 3.3.1

Realm sync feature enabled: no

Android Studio version: 2.3.3

Which Android version and device: Galaxy Note 5 Android 6.0.1

**Comment by Zhuinden** [comment]
This core PR got merged 3 hours ago which allows this use case to work, see https://github.com/realm/realm-core/pull/2644

This means it'll need to be included in object store and then exposed in Java binding (probably just removing depreciation will be enough though)

**Comment by ribbon-xx** [comment]
Thanks for your reply, @Zhuinden. 
Will it be released in next release (3.3.2 maybe)?
And do you know when is it?

Regards,

**Comment by Zhuinden** [comment]
Tricky question because core needs to be released.

Current release is 2.8.3, and this would probably be in 2.9.0, which will need to show up in Realm-Java...

**Comment by zaki50** [comment]
@ribbon-xx Does the following code work? I inserted `.where()` before `distinct()`.

```
RealmResults<MyModel> realmResults = getRealm().where(MyModel.class)
					.notEqualTo(MyModel.GROUP_ID, 0)
					.notEqualTo(MyModel.IS_DELETED, 0)
					.findAllSorted(MyModel.LAST_UPDATE, Sort.DESCENDING);
			realmResults.where().distinct(MyModel.GROUP_ID);
```

**Comment by Zhuinden** [comment]
@zaki50 the problem is that distinct used the original table view and not the queried one, the core support just got merged afaik

**Comment by kneth** [comment]
The new behaviour of `distinct()` is a breaking change. So I think we need to wait until next major release to include it.

**Comment by Zhuinden** [comment]
@kneth technically it's not "more breaking" than 

> 3.1.0 (2017-04-05)
> findFirstAsync() now returns an invalid object if there is no object matches the query condition instead of running the query repeatedly until it can find one (#4352).

**Comment by Zhuinden** [comment]
``` java    
realm.where(MyModel.class)
					.notEqualTo(MyModel.GROUP_ID, 0)
					.notEqualTo(MyModel.IS_DELETED, 0)
					.findAllSorted(MyModel.LAST_UPDATE, Sort.DESCENDING)
                                        .where()
                                        .distinct(MyModel.GROUP_ID);
```

will work in Realm-Java 4.0

**Comment by cmelchior** [comment]
Yes, this has been fixed in Realm Java 4.0.0
